### DISCUSSION TITLE
POLL: Default URL Bar Preference

### DISCUSSION THREAD

**Discussion by Lokke** [zen-browser/desktop#3686] [Polls]
After reading through GitHub issues, it seems the floating URL bar in Zen has been a hot topic, with many users unhappy about the change. I couldn’t find anyone praising it—so I’m curious: does anyone actually like it? Should Zen switch back to the classic URL bar by default? Let’s discuss!

**Comment by Laflamme** [comment]
The floating URL bar makes the most sense in the context of the narrow single toolbar. I believe it should be on by default in that context and off by default otherwise.

With that being said, the execution of the floating toolbar could be improved. Its main problem, which is not specific to Zen Browser, but made worse with the single toolbar, is that you must click on the upper left to interact with something that will appear in the center of the window, which can be disorienting.

When invoked by a keyboard shortcut, the floating URL bar should remain where it currently is. But when invoked by a click, I believe it should extend from the URL bar in the single toolbar -- "extend" as in "let's use an animation to extend the field, and keep it anchored with the single toolbar URL bar".

What do you think?

Edit: Actually, maybe we could add the option to have the regular URL toolbar when we click on it, but the floating URL bar when we invoke it with a keyboard shortcut. The issues with the URL bar being too crowded in the context of the narrow single toolbar could be mitigated by making it two rows high, with the actual URL on top row  and its controls (padlock, zoom, star, etc.) on the second row below.

**Comment by VoxelPrismatic** [comment]
This isn't centered to anything at all, and more importantly, double-clicking the text to select a specific part of the URL no longer works.

You are welcome to use the new floating URL bar on new tabs/windows, but when I click the text box, the text box should not move.
![image](https://github.com/user-attachments/assets/eb4a0c20-3100-44b8-b1b2-37783c3d4250)

**Comment by ghalestrilo** [comment]
I understand there must have been a ton of research going into the design of this feature, but it's a big usability issue to have an input field jumping around while the user tries to type - especially if the whole point of Zen is to provide a calmer, smoother, more focused experience. This may not represent a wide audience, but currently the feature is so distracting I struggle to even use the browser. The lack of an option to turn it off is a dealbreaker. I put in a vote and look forward to the outcome.

**Comment by ghalestrilo** [comment]
This is no longer happening as of the latest updates. Even tough it's a dead thread, we have to share love and appreciation for the team behind Zen for responding this quickly to feedback. It reflects their level of attention and commitment to the users. Great stuff! And as a user thanks for developing the best browser out there ATM. It's been a great experience to use it!
